src/main/java/pl/futurecollars/invoicing/controllers/TaxCalculatorController.java

```diff
@@ -22,18 +19,10 @@
 
     private final TaxCalculatorService taxCalculatorService;
 
-    @GetMapping(path = "/tax/{taxIdentificationNumber}", produces = {"application/json;charset=UTF-8"})
-    public ResponseEntity<TaxReport> getTaxReport(@PathVariable String taxIdentificationNumber) {
-        log.debug("Generate tax report for company with Tax Identification Number: " + taxIdentificationNumber);
-        return ResponseEntity.ok()
-            .body(taxCalculatorService.taxReport(taxIdentificationNumber));
-    }
-
-    @ExceptionHandler(Exception.class)
-    public ResponseEntity<Json> handleError(HttpServletRequest req, Exception ex) {
-        log.error("Request: " + req.getRequestURL() + " raised " + ex);
-
-        return ResponseEntity.badRequest().lastModified(ZonedDateTime.now()).build();
+    @PostMapping(path = "/tax", produces = {"application/json;charset=UTF-8"})
+    public ResponseEntity<TaxReport> getTaxReport(@RequestBody Company company) {
+        log.debug("Generate tax report for company with Tax Identification Number: " + company.getTaxIdentificationNumber());
```

Comment:
try to use {} as a "parameter" in logger. e.g.
log.debug("Generate tax report for company with Tax Identification Number: {} ", company.getTaxIdentificationNumber());